src/arch/x86/idle.c

```diff
@@ -7,11 +7,18 @@
 #include <config.h>
 #include <api/debug.h>
 
-void idle_thread(void)
+/*
+ * The idle thread does not have a dedicated stack and runs in
+ * the context of the idle thread TCB. Make sure that the compiler
+ * always eliminates the function prologue by declaring the
+ * idle_thread with the naked attribute.
+ */
+__attribute__((naked)) void idle_thread(void)
 {
-    while (1) {
-        asm volatile("hlt");
-    }
+    asm volatile(
```

Comment:
Can you add a warning here that a `for(;;) { ... }` or ``while(1) { ... }` is also dangerous because the compiler creates additional code then that might need stack.

Reply:
Ok yeah I can do that. The compiler would not allow any `for` or `while` statement in a naked function anyway --- the function body of a naked function must be one big `asm` statement.

Reply:
Done!

Reply:
This seems a way to write plain asm code in a C file also without the pitfalls of `naked` (only simple asm, no parameters...), but I wonder if the C parser used in verification can hadle this.
 
```
asm("                     \n\
  _asm_idle_thread:       \n\
    hlt                   \n\
	jmp _asm_idle_thread  \n\
");
int idle_thread() asm ("_asm_idle_thread"); 

int kernel_has_nothing_to_do() {
        idle_thread();
        return 0;
}
```

Using a dedicated `idle.s` might be the better approach then?

Reply:
We would need two `idle.s` files (one for `x86` one for `x86_64`) that are identical so there might be a duplication of code problem there. The `naked` attribute seems to be the clearest way of expressing the intention here.